File: lambdas/assessment/entity/quiz.py
class Quiz:
    def __init__(self, difficulty, topic, subtopic, description, is_original, id=None, context=None, created_at=None):
        self.id = id
        self.context = context
        self.created_at = str(created_at)

        if difficulty is None:
            raise ValueError('difficulty is required')
        self.difficulty = difficulty
        if topic is None:
            raise ValueError('topic is required')
        self.topic = topic
        if subtopic is None:
            raise ValueError('subtopic is required')
        self.subtopic = subtopic
        if description is None:
            raise ValueError('description is required')
        self.description = description
        if is_original is None:
            raise ValueError('is_original is required')
        self.is_original = is_original

    def to_dict(self):
        return {
            'id': self.id,
            'context': self.context,
            'difficulty': self.difficulty,
            'topic': self.topic,
            'subtopic': self.subtopic,
            'description': self.description,
            'is_original': self.is_original,
            'created_at': self.created_at
        }

File: web/src/data/quiz.ts
import { Quiz } from '@/types';

export const mockedSelectedQuiz: Quiz = {
  id: '1',
  context:
    'How would you optimize a Dockerfile for a web application to ensure efficient build times and image sizes? Describe the steps you would take and the rationale behind them.',
  description: 'Optimizing Dockerfile',
  topic: 'typescript',
  subtopic: 'docker',
  isOriginal: true,
  createdAt: '2024-02-08 23:02:57.478524',
  questions: [
    {
      id: '1',
      text: 'How would you optimize a Dockerfile for a web application to ensure efficient build times and image sizes? Describe the steps you would take and the rationale behind them.',
      questionNumber: 1,
    },
    {
      id: '2',
      text: 'What are the steps you would take to optimize a Dockerfile for a web application to ensure efficient build times and image sizes? Describe the rationale behind them.',
      questionNumber: 2,
    },
    {
      id: '3',
      text: 'Describe the steps you would take to optimize a Dockerfile for a web application to ensure efficient build times and image sizes.',
      questionNumber: 3,
    },
  ],
  difficulty: 'medium',
};

export const mockedQuizzes: Quiz[] = [
  mockedSelectedQuiz,
  {
    id: '2',
    context: 'context here',
    description: 'Description 2',
    topic: 'Topic 2',
    subtopic: 'Subtopic 2',
    difficulty: 'Medium',
    isOriginal: true,
    questions: [
      {
        id: '1',
        text: 'Question 1',
        questionNumber: 1,
      },
      {
        id: '2',
        text: 'Question 2',
        questionNumber: 2,
      },
      {
        id: '3',
        text: 'Question 3',
        questionNumber: 3,
      },
    ],
    createdAt: '2024-02-08 23:02:57.478524',
  },
  {
    id: '3',
    context: 'context here',
    description: 'Description 3',
    topic: 'Topic 3',
    subtopic: 'Subtopic 3',
    difficulty: 'Hard',
    isOriginal: true,
    questions: [
      {
        id: '1',
        text: 'Question 1',
        questionNumber: 1,
      },
      {
        id: '2',
        text: 'Question 2',
        questionNumber: 2,
      },
      {
        id: '3',
        text: 'Question 3',
        questionNumber: 3,
      },
    ],
    createdAt: '2024-02-08 23:02:57.478524',
  },
];
